kvserver: small changes

wenyihu6 opened the pull request:
Epic: none 
Release note: none 

----

**allocator: move isDecommissionAction to allocatorimpl**

This commit refactors isDecommissionAction into allocatorimpl for consistency
with other similar helpers like allocatorActions.{Add,Replace,Remove}. This
change has no behavior changes but to make future commits easier.

---

**kvserver: simplify logging in maybeEnqueueProblemRange**

This commit simplifies the logging in `maybeEnqueueProblemRange` to log two
booleans directly.

---

**kvserver: fix comment when dropping due to exceeding size**

Previously, the comment on the queue incorrectly stated that it removes the
lowest-priority element when exceeding its maximum size. This was misleading
because heap only guarantees that the root is the highest priority, not that
elements are globally ordered. This commit updates the comment to clarify that
the removed element might not be the lowest priority. Ideally, we should drop
the lowest priority element when exceeding queue size, but heap doesn't make
this very easy.

---

**kvserver: add logging for ranges dropped from base queue**

This commit adds logging for ranges dropped from the base queue due to exceeding
max size, improving observability. The log is gated behind V(1) to avoid
verbosity on nodes with many ranges.

wenyihu6 committed:
allocator: move isDecommissionAction to allocatorimpl

This commit refactors isDecommissionAction into allocatorimpl for consistency
with other similar helpers like allocatorActions.{Add,Replace,Remove}. This
change has no behavior changes but to make future commits easier.

wenyihu6 committed:
kvserver: fix comment when dropping due to exceeding size

Previously, the comment on the queue incorrectly stated that it removes the
lowest-priority element when exceeding its maximum size. This was misleading
because heap only guarantees that the root is the highest priority, not that
elements are globally ordered. This commit updates the comment to clarify that
the removed element might not be the lowest priority. Ideally, we should drop
the lowest priority element when exceeding queue size, but heap doesn't make
this very easy.

wenyihu6 committed:
kvserver: simplify logging in maybeEnqueueProblemRange

This commit simplifies the logging in `maybeEnqueueProblemRange` to log two
booleans directly.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152508)
<!-- Reviewable:end -->

wenyihu6 committed:
kvserver: add logging for ranges dropped from base queue

This commit adds logging for ranges dropped from the base queue due to exceeding
max size, improving observability. The log is gated behind V(1) to avoid
verbosity on nodes with many ranges.

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo comments

> @arulajmani reviewed 2 of 2 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152508#-:-OYbsjwm3qAEgdn5p6Zy:b-913mr5)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[`-- commits` line 33 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/152508#-OYbsg4J120vt6-_61DJ:-OYbsg4J120vt6-_61DK:bvzsluv):*
Did you mean to keep this fixup?
___
*[`pkg/kv/kvserver/queue.go` line 776 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/152508#-OYbsUQF8mjtCQQOM1hW:-OYbsUQF8mjtCQQOM1hX:b67rztv) ([raw file](https://github.com/cockroachdb/cockroach/blob/5579035536b01fb52758b2870cab14637ba26505/pkg/kv/kvserver/queue.go#L776)):*
> ```Go
> 	if pqLen := bq.mu.priorityQ.Len(); pqLen > bq.maxSize {
> 		replicaItemToDrop := bq.mu.priorityQ.sl[pqLen-1]
> 		log.Dev.VInfof(ctx, 1, "dropping due to exceeding queue max size: priority=%0.3f, replica=%v",
> ```

nit: Should this be in KVDistribution?


<!-- Sent from Reviewable.io -->

wenyihu6 commented:
*[`pkg/kv/kvserver/queue.go` line 776 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/152508#-OYbsUQF8mjtCQQOM1hW:-OYbul-g4um51ef86uVT:brqp11y) ([raw file](https://github.com/cockroachdb/cockroach/blob/5579035536b01fb52758b2870cab14637ba26505/pkg/kv/kvserver/queue.go#L776)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: Should this be in KVDistribution?
</blockquote></details>

Hm, was wondering abt this as well. But other logs in this file use `log.Dev`, and this is for base queue which is used for other queues outside of kv distribution, such as mvccGC. 


<!-- Sent from Reviewable.io -->

wenyihu6 commented:
TFTR!

bors r=arulajmani
